### ISSUE TITLE
Wrong result when rendering multiple volumes

### ISSUE THREAD

**Issue by wilsonCernWq** [ospray/ospray#134]
Hi

I found OSPRay will give wrong rendering result when there are multiple volumes in the scene. What I did was simply putting four identical volumes next to each other, and rendered them using one ospModel.

In the result, some volumes seemed to be skipped by the ray and became transparent (volumes supposed to be visible were missed). Look at areas near volume boundaries in the attached images.

I tested my code with both block_bricked_volume and shared_structured_volume. They gave me identical results. My code is uploaded here: https://github.com/wilsonCernWq/OSPRayTinyLoader/blob/volume-debug/src/otv_main.cpp

![screenshot from 2017-03-16 16-18-07](https://cloud.githubusercontent.com/assets/13118582/24024961/ecd16b2a-0a7c-11e7-9861-04d7b712c75e.png)

![screenshot from 2017-03-16 16-18-53](https://cloud.githubusercontent.com/assets/13118582/24024967/f90f9a1a-0a7c-11e7-9eb4-49d8060b0167.png)

![screenshot from 2017-03-16 16-17-30](https://cloud.githubusercontent.com/assets/13118582/24024982/14898792-0a7d-11e7-8330-ed15bb6350b6.png)

Can anyone help me with that? Thanks!

**Comment by johguenther** [comment]
Hi,
you are right: multiple volumes are currently not well handled in OSPRay (even more so when they intersect). The SciVis renderer is about to be refactored, also with the goal to better handle your situation. We'll keep you posted when the improvements are available.
Best regards,
  Johannes

**Comment by wilsonCernWq** [comment]
Hi
I noticed my issue is here still.
Actually I found the bug a while ago already and updated it inside my fork
I just want to make this as a reminder so that the issue will be fixed in the next release :smile: 

In ospray/ospray/render/scivis/SciVisRenderer.ispc line 212
```
            volumeColor = SciVisRenderer_computeVolumeInterval(renderer, v, ray, tBegin, tEnd, .98f - color.w, 0, rayOffset, sampleID, 1.f);
```
The 6th argument should be changed to 1.0f or 0.99f. The early termination opacity should always be one for each volume. Otherwise the ray will exit too early and go directly into the second volume.
